src/analysis/normed_space/weak_dual.lean

```diff
@@ -0,0 +1,131 @@
+consequence the weak-* topology is coarser than the topology induced by the dual norm (operator
+norm).
+
+## Main definitions
+
+The main definitions concern the canonical mapping `dual 𝕜 E → weak_dual 𝕜 E`.
+
+* `to_weak_dual` is a linear equivalence from `dual 𝕜 E`to `weak_dual 𝕜 E`.
+* `continuous_linear_map_to_weak_dual` is a continuous linear mapping from
+  `dual 𝕜 E` to `weak_dual 𝕜 E`.
+
+## Main results
+
+The file is a stub.
+
+The first main results concern the comparison of the operator norm topology on `dual 𝕜 E` and the
+weak-* topology on (it type synonym) `weak_dual 𝕜 E`:
+* `dual_norm_topology_le_weak_dual_topology` is the statement that the weak-* topology on the dual
+  of a normed space is coarser (not necessarily strictly) than the operator norm topology.
+
+TODOs:
+* Add Banach-Alaoglu theorem.
+* Add metrizability of the dual unit ball (more generally bounded subsets) of `weak_dual 𝕜 E`
+  under the assumption of separability of `E`. This would in particular give the sequential
+  Banach-Alaoglu theorem from the general one.
+
+## Notations
+
+No new notation is introduced.
+
+## Implementation notes
+
+Weak-* topology is defined generally in the file <topology.algebra.weak_dual_topology>.
+
+When `E` is a normed space, the duals `dual 𝕜 E` and `weak_dual 𝕜 E` are type synonyms with
+different topology instances.
+
+## References
+
+* https://en.wikipedia.org/wiki/Weak_topology#Weak-*_topology
+
+## Tags
+
+weak-star, weak dual
+
+-/
+
+noncomputable theory
+open filter
+open_locale topological_space
+
+section weak_star_topology_for_duals_of_normed_spaces
+/-!
+### Weak star topology on duals of normed spaces
+In this section, we prove properties about the weak-* topology on duals of normed spaces.
+We prove in particular that the canonical mapping `dual 𝕜 E → weak_dual 𝕜 E` is continuous,
+i.e., that the weak-* topology is coarser (not necessarily strictly) than the topology given
+by the dual-norm (i.e. the operator-norm).
+-/
+
+open normed_space
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+variables {E : Type*} [normed_group E] [normed_space 𝕜 E]
+
+/-- For normed spaces `E`, there is a canonical map `dual 𝕜 E → weak_dual 𝕜 E` (the "identity"
+    mapping). It is a linear equivalence. -/
+def normed_space.dual.to_weak_dual : dual 𝕜 E ≃ₗ[𝕜] weak_dual 𝕜 E :=
+linear_equiv.refl 𝕜 (E →L[𝕜] 𝕜)
+
+/-- For normed spaces `E`, there is a canonical map `weak_dual 𝕜 E → dual 𝕜 E` (the "identity"
+    mapping). It is a linear equivalence. Here it is implemented as the inverse of the linear
+    equivalence `normed_space.dual.to_weak_dual` in the other direction. -/
+def weak_dual.to_normed_dual : weak_dual 𝕜 E ≃ₗ[𝕜] dual 𝕜 E :=
+normed_space.dual.to_weak_dual.symm
+
+@[simp] lemma weak_dual.coe_to_fun_eq_normed_coe_to_fun (x' : dual 𝕜 E) :
+  (x'.to_weak_dual : E → 𝕜) = x' := rfl
+
+namespace normed_space.dual
+
+@[simp] lemma to_weak_dual_inj_iff (x' y' : dual 𝕜 E) :
```

Comment:
Do you mind renaming this lemma from `inj_iff` to `eq_iff`, and same for the next, and using the `injective.eq_iff` proof for the next one also?  Sorry, it's pretty minor, but it helps to be consistent about these things (as the discussion with Eric on the other lemma shows ...)

Reply:
I will change this. I'll just first wait and see that the checks pass for the first time :crossed_fingers:.

Reply:
Actually, I realized that the `.injective.eq_iff` dot-notation is so convenient that I wonder if these lemmas are worth having at all. (Again an idea I wasn't sufficiently actively familiar with, so I initially thought these simp-lemmas might be practical.)

I'll rename for now, but might actually just remove, too, depending on what you think.

Reply:
I count some 40 occurrences of `[something].injective.eq_iff` in the library, of which about 10 are uses like yours (recording the pattern as a stand-alone simp-lemma) and about 30 are uses within a bigger proof.  So there's no particular consensus .... let's keep it, it can do no harm!